app/src/main/java/com/example/grocerieswizard/ui/home/HomeFragment.java

```diff
@@ -249,4 +174,73 @@ private String getStringIngredients(List<IngredientUi> ingredients) {
         return stringBuilder.toString();
     }
 
+    @Override
+    public void showRecipes(List<RecipeUi> recipes) {
+        adapter.setRecipeList(recipes);
+    }
+
+    private void showRecipeDetails(RecipeUi recipe) {
+        if (recipe != null) {
+            if (!recipe.isSwiped()) {
+                recipe.setSwiped(false);
+                adapter.itemChanged(adapter.getPositionForRecipe(recipe));
+                DetailFragment detailFragment = DetailFragment.newInstance(recipe);
+                requireActivity().getSupportFragmentManager()
+                        .beginTransaction()
+                        .replace(R.id.frameLayout, detailFragment)
+                        .addToBackStack(null)
+                        .commit();
+            } else {
+                Log.d("MainActivity", "recipe model null");
+            }
+        }
+    }
+
+    @Override
+    public void showDeleteConfirmation(RecipeUi recipe) {
+            AlertDialog.Builder builder = new AlertDialog.Builder(requireContext());
+            builder.setTitle("Confirm Deletion");
+            builder.setMessage("Are you sure you want to delete " + recipe.getRecipeName() + " recipe?");
+            builder.setPositiveButton("Delete", (dialog, which) -> {
+                // User confirmed deletion, remove the recipe and update the RecyclerView
+                adapter.removeRecipe(recipe);
```

Comment:
For future. Where is this recipe deleted from? It seems like it is only deleted from the screen, not from database.

Reply:
homePresenter has this:
`    public void deleteRecipe(RecipeUi recipeUi) {
        recipeRepository.deleteRecipe(recipeUi);
        view.showDeleteConfirmation(recipeUi);
    }`

but this leads a new bug. when the user clicks cancel during the approve the deletion, it doesnt delete from the screen at that moment but when you reload that screen at some point it has seen as removed. because  recipeRepository.deleteRecipe(recipeUi)  responsible for deletion from db :(